File: generate_app_icon.js
#!/usr/bin/env node
/**
 * Generate WalkMyPet app icon with paw logo
 * Uses node-canvas to create a gradient background with paw print
 */

const fs = require('fs');
const path = require('path');

// Try to load canvas, install if needed
let Canvas;
try {
  Canvas = require('canvas');
} catch (e) {
  console.log('Installing canvas package...');
  require('child_process').execSync('npm install canvas', { stdio: 'inherit' });
  Canvas = require('canvas');
}

const { createCanvas } = Canvas;

function createGradient(ctx, width, height, color1, color2) {
  const gradient = ctx.createLinearGradient(0, 0, width, height);
  gradient.addColorStop(0, color1);
  gradient.addColorStop(1, color2);
  return gradient;
}

function drawPaw(ctx, cx, cy, size, color) {
  ctx.fillStyle = color;

  // Main pad (larger, bottom)
  const padWidth = size * 0.6;
  const padHeight = size * 0.5;

  ctx.beginPath();
  ctx.ellipse(
    cx,
    cy + size * 0.4,
    padWidth / 2,
    padHeight / 2,
    0,
    0,
    Math.PI * 2
  );
  ctx.fill();

  // Toe pads (4 smaller circles above main pad)
  const toeSize = size * 0.22;
  const toePositions = [
    { x: cx - size * 0.35, y: cy - size * 0.15 },  // Left toe
    { x: cx - size * 0.12, y: cy - size * 0.35 },  // Left-center toe
    { x: cx + size * 0.12, y: cy - size * 0.35 },  // Right-center toe
    { x: cx + size * 0.35, y: cy - size * 0.15 },  // Right toe
  ];

  toePositions.forEach(pos => {
    ctx.beginPath();
    ctx.arc(pos.x, pos.y, toeSize / 2, 0, Math.PI * 2);
    ctx.fill();
  });
}

function createIcon(size, foregroundOnly = false) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  if (!foregroundOnly) {
    // Gradient background: #6366F1 (indigo) to #8B5CF6 (purple)
    const gradient = createGradient(
      ctx,
      size,
      size,
      'rgb(99, 102, 241)',
      'rgb(139, 92, 246)'
    );
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, size, size);
  }

  // Draw white paw in center
  const centerX = size / 2;
  const centerY = size / 2;
  const pawSize = size * 0.45;

  drawPaw(ctx, centerX, centerY, pawSize, 'white');

  return canvas;
}

async function main() {
  console.log('🐾 Generating WalkMyPet app icons...');

  // Ensure assets/icon directory exists
  const iconDir = path.join(__dirname, 'assets', 'icon');
  if (!fs.existsSync(iconDir)) {
    fs.mkdirSync(iconDir, { recursive: true });
  }

  // Create main icon (1024x1024 for iOS, will be resized by flutter_launcher_icons)
  console.log('Creating main icon (1024x1024)...');
  const icon1024 = createIcon(1024);
  const iconPath = path.join(iconDir, 'app_icon.png');
  const buffer = icon1024.toBuffer('image/png');
  fs.writeFileSync(iconPath, buffer);

  // Create adaptive icon foreground (transparent background)
  console.log('Creating adaptive icon foreground (1024x1024)...');
  const foreground = createIcon(1024, true);
  const foregroundPath = path.join(iconDir, 'app_icon_foreground.png');
  const foregroundBuffer = foreground.toBuffer('image/png');
  fs.writeFileSync(foregroundPath, foregroundBuffer);

  console.log('✅ Icons generated successfully!');
  console.log('📁 Saved to:');
  console.log(`   - ${iconPath}`);
  console.log(`   - ${foregroundPath}`);
  console.log('\nNext step: Run "dart run flutter_launcher_icons"');
}

main().catch(err => {
  console.error('Error:', err);
  process.exit(1);
});

File: generate_app_icon.py
#!/usr/bin/env python3
"""
Generate WalkMyPet app icon with paw logo
Uses PIL/Pillow to create a gradient background with paw print
"""

try:
    from PIL import Image, ImageDraw
    import math
except ImportError:
    print("Installing required packages...")
    import subprocess
    subprocess.check_call(['pip3', 'install', 'pillow'])
    from PIL import Image, ImageDraw
    import math


def create_gradient(width, height, color1, color2):
    """Create a diagonal gradient from top-left to bottom-right"""
    base = Image.new('RGB', (width, height), color1)
    top = Image.new('RGB', (width, height), color2)
    mask = Image.new('L', (width, height))
    mask_data = []
    for y in range(height):
        for x in range(width):
            # Diagonal gradient
            distance = math.sqrt(x**2 + y**2) / math.sqrt(width**2 + height**2)
            mask_data.append(int(255 * distance))
    mask.putdata(mask_data)
    base.paste(top, (0, 0), mask)
    return base


def draw_paw(draw, cx, cy, size, color):
    """Draw a paw print with main pad and 4 toe pads"""
    # Main pad (larger, bottom)
    pad_width = size * 0.6
    pad_height = size * 0.5
    main_pad = [
        (cx - pad_width/2, cy + size*0.15),
        (cx + pad_width/2, cy + size*0.15),
        (cx + pad_width/2, cy + size*0.55),
        (cx - pad_width/2, cy + size*0.55),
    ]
    draw.ellipse([
        cx - pad_width/2, cy + size*0.15,
        cx + pad_width/2, cy + size*0.65
    ], fill=color)

    # Toe pads (4 smaller ovals above main pad)
    toe_size = size * 0.22
    toe_positions = [
        (cx - size*0.35, cy - size*0.15),  # Left toe
        (cx - size*0.12, cy - size*0.35),  # Left-center toe
        (cx + size*0.12, cy - size*0.35),  # Right-center toe
        (cx + size*0.35, cy - size*0.15),  # Right toe
    ]

    for tx, ty in toe_positions:
        draw.ellipse([
            tx - toe_size/2, ty - toe_size/2,
            tx + toe_size/2, ty + toe_size/2
        ], fill=color)


def create_icon(size, output_path, foreground_only=False):
    """Create app icon with gradient background and white paw"""
    if foreground_only:
        # Transparent background for adaptive icon foreground
        img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    else:
        # Gradient background: #6366F1 (indigo) to #8B5CF6 (purple)
        img = create_gradient(size, size, (99, 102, 241), (139, 92, 246))
        img = img.convert('RGBA')

    draw = ImageDraw.Draw(img)

    # Draw white paw in center
    center_x = size / 2
    center_y = size / 2
    paw_size = size * 0.45

    draw_paw(draw, center_x, center_y, paw_size, (255, 255, 255, 255))

    return img


def main():
    print("🐾 Generating WalkMyPet app icons...")

    # Create main icon (1024x1024 for iOS, will be resized by flutter_launcher_icons)
    print("Creating main icon (1024x1024)...")
    icon_1024 = create_icon(1024, "assets/icon/app_icon.png")
    icon_1024.save("assets/icon/app_icon.png", "PNG")

    # Create adaptive icon foreground (transparent background)
    print("Creating adaptive icon foreground (1024x1024)...")
    foreground = create_icon(1024, "assets/icon/app_icon_foreground.png", foreground_only=True)
    foreground.save("assets/icon/app_icon_foreground.png", "PNG")

    print("✅ Icons generated successfully!")
    print("📁 Saved to:")
    print("   - assets/icon/app_icon.png")
    print("   - assets/icon/app_icon_foreground.png")
    print("\nNext step: Run 'flutter pub get' then 'dart run flutter_launcher_icons'")


if __name__ == "__main__":
    main()
